pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -91,7 +94,12 @@ func (n *Node) Add(pod *v1.Pod) error {
 
 	// Check instance type combinations
 	requests := resources.Merge(n.requests, resources.RequestsForPods(pod))
-	instanceTypes := filterInstanceTypes(n.InstanceTypeOptions, nodeRequirements, requests)
+	// determine the number of volumes that will be mounted if the pod schedules
+	mountedVolumeCount, err := n.volumeUsage.Validate(pod)
+	if err != nil {
+		return err
+	}
+	instanceTypes := filterInstanceTypes(n.InstanceTypeOptions, nodeRequirements, requests, mountedVolumeCount)
```

Comment:
I'd expect the volume count to be modeled using the resource requests.

Reply:
They're pretty different.  Pods don't have volumes in their resource requests, they are implicit based off of volumes that are mounted.  They are also shared, ten pods can use the same PVC which will reduce the number of volumes by one since only one volume has been mounted.

Reply:
Is `node.capacity[ebs...]` simply node used then?